bokehjs/src/coffee/models/renderers/glyph_renderer.coffee

```diff
@@ -157,7 +157,7 @@ export class GlyphRendererView extends RendererView
       else if selected['1d'].indices.length > 0
         selected = selected['1d'].indices
       else
-        selected = []
+        selected = (parseInt(i) for i in Object.keys(selected["2d"].indices))
```

Comment:
because ``Object.keys`` casts objects keys to strings. Or, more precisely, it returns the internally stored values (which are strings) instead of casting back the dtypes (I think).